woocommerce/woocommerce-android
# Cache account settings

Closes: #13198

### Description
This PR improves the account settings performance by saving the settings on the local preferences and displaying that information when available instead of only relying on the API request. With the changes introduced here, the local settings preferences will be the source of truth, and every time we fetch the account settings, we will update the preferences value.

### Testing information
TC1
1. Open the orders section
2. Open an order eligible for shipping label creation
3. Tap on create shipping label
4. Check that the account settings are fetched from the API
5. Navigate back
6. Tap again on create shipping label
7. Check that the account settings are presented from the preferences
8. Check that the account settings are refreshed on background

TC2
1. Open the orders section
2. Open an order eligible for shipping label creation
3. Tap on create shipping label
4. Check that the account settings are fetched from the API
5. Check that when the API request fails then the error screen is displayed

* You can apply the below path to force the fetch settings to fail
<details>
  <summary>Patch</summary>

```
Index: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/FetchAccountSettings.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/FetchAccountSettings.kt b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/FetchAccountSettings.kt
--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/FetchAccountSettings.kt	(revision afcf87f039221212a273e4c3a6b410a4424ce430)
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/FetchAccountSettings.kt	(date 1735139698227)
@@ -10,19 +10,6 @@
     private val selectedSite: SelectedSite
 ) {
     suspend operator fun invoke(): Result<StoreOptionsModel> {
-        return selectedSite.getOrNull()?.let {
-            val response = shippingRepository.fetchAccountSettings(it)
-            val result = response.model
-            when {
-                response.isError.not() && result != null && result != StoreOptionsModel.EMPTY -> {
-                    Result.success(result)
-                }
-
-                else -> {
-                    val message = response.error?.message ?: "Unknown error"
-                    Result.failure(Exception(message))
-                }
-            }
-        } ?: Result.failure(Exception("No site selected"))
+        return Result.failure(Exception("No site selected"))
     }
 }

```

</details>

### The tests that have been performed
- Tested that the account settings are fetched from the preferences
- Checked that an API request is made to refresh the settings data
- Checked that when the request fails an error is displayed

### Images/gif

https://github.com/user-attachments/assets/754f7565-7877-4123-9189-0971569f63e1


## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
